src/types/tuple.rs

```diff
@@ -213,6 +214,15 @@ impl PyTuple {
             length: self.len(),
         }
     }
+
+    /// Return a new list containing the contents of this tuple; equivalent to the Python expression `list(tuple)`.
+    ///
+    /// This method is equivalent to `self.as_sequence().list()` and faster than `PyList::new(py, self)`.
+    pub fn to_list(&self) -> &PyList {
+        self.as_sequence()
+            .list()
+            .expect("failed to convert tuple to list")
+    }
```

Comment:
(I think) you can do  `PyList::new(py, self.as_slice())` on `not(Py_LIMITED_API)`

Reply:
Oh great idea, I'll try that when I get a free moment and see how it performs...

Reply:
Hmm unfortunately that performs very similarly to `PyList::new(py, self)` (i.e. slower than going through the sequence API), so the bottleneck must be in constructing the list on the Rust side. An investigation for another time, perhaps...